### ISSUE TITLE
How use NukeVideo with SwiftUI

### ISSUE THREAD

**Issue by dumojo** [kean/Nuke#688]

**Comment by kean** [comment]
It can be used with [`LazyImage`](https://kean-docs.github.io/nukeui/documentation/nukeui/lazyimage).

```swift
LazyImage(url: URL(string: "https://example.com/image.jpeg")) { state in
    if let container = state.imageContainer {
        if let data = container.data, container.type?.isVideo ?? false {
            // Use `VideoPlayerView` wrapped in `UIViewRepresentable`
        } else {
            state.image // Use the default view
        }
    }
}
```

For more info, please follow the [migration notes](https://github.com/kean/Nuke/blob/main/Documentation/Migrations/Nuke%2012%20Migration%20Guide.md).

I'm planning to add a `UIViewRepresentable` to the framework itself in the future.

**Comment by itamargilvybe** [comment]
So there's no "simple" way to show a video? I used to do this but it doesn't seem to work no more:
LazyImage(url: url) { state in
                if let image = state.image {
                    image
                        .resizable()
                        .scaledToFill()
                } else if state.error != nil { // Always goes in here
                    Text("😵‍💫")
                } else {
                    ProgressView().progressViewStyle(.circular)
                }
            }

url for testing - https://firebasestorage.googleapis.com:443/v0/b/vybetogether.appspot.com/o/vybes%2F4D6D55BE-C0C3-4560-8E86-5E0CE5F5B24B%2Fmedia1.mov?alt=media&token=998f6457-6c51-47dd-be6c-d2ab3678225a

**Comment by kean** [comment]
There was an issue in `12.0` related to video. It's fixed in [12.1](https://github.com/kean/Nuke/releases/tag/12.1.0). But yes, this is how it's going to work moving forward. The idea is to make the core framework as small as possible and provide users the full power to customize both the loading and the displaying of images.
